crml/eth-state-oracle/src/lib.rs

```diff
@@ -365,18 +398,78 @@ decl_module! {
 			}
 		}
 
+		/// Deposits a bond which is required to submit challenges
+		/// call requests can't be made if there are no challengers available to challenge them
+		#[weight = 500_000]
+		pub fn bond_challenger(origin) {
+			let origin = ensure_signed(origin)?;
+
+			// Check account doesn't already have a bond
+			if !Self::challenger_bonds(&origin).is_zero() {
+				// Account already has CPay bonded
+				return Err(Error::<T>::AlreadyBonded.into())
+			};
+
+			// check user has the requisite funds to make this bond
+			let fee_currency = T::MultiCurrency::fee_currency();
+			let challenger_bond_amount = T::ChallengerBondAmount::get();
+			let max_concurrent_responses = Self::max_concurrent_responses();
```

Comment:
this value should be calculated
```suggestion
			let max_concurrent_responses = T::MaxRequestsPerBlock::get() * T::ChallengePeriod::get() / MaxRelayerCount;
```